docs(changelog): for release v14.0.0 [skip ci] (#857)
commercetools/merchant-center-application-kit: CHANGELOG.md

## [14.0.0](https://github.com/commercetools/merchant-center-application-kit/compare/v13.13.3...v14.0.0) (2019-07-18)

> This release introduces several **breaking changes**, as documented below, mostly around React Hooks and permissions.

Reviewer: ```suggestion
This release introduces several **breaking changes**, as documented below, mostly around React Hooks and our own permission functionality.
```

> You can read the v14 announcement and version tracker here: https://github.com/commercetools/merchant-center-application-kit/issues/748

### Full support for React Hooks

> We bumped some peer dependencies that provide full hooks support, so you need to update those:

Reviewer: ```suggestion
In order to support hooks the following peer dependencies have to be updated:
```

- `react>=16.8`
- `react-dom>=16.8`
- `react-intl>=3` (_currently still in beta_)
- `react-redux>=7`

> _If you're not familiar with React Hooks, we recommend to have a look at the following resources, as Hooks are now the recommended way to write components:_

Reviewer: ```suggestion
_If need more information about React Hooks, we recommend to have a look at the following resources. Hooks are another way to write components and come with many benefits:_
```

- https://reactjs.org/docs/hooks-intro.html
- https://egghead.io/courses/simplify-react-apps-with-react-hooks
- https://usehooks.com/

> _TL;DR; using Hooks makes the code more readable, easier to test, less code to ship_

Reviewer: ```suggestion
_TL;DR; with Hooks code can become more readable, easier to test while often improving performance and reducing bundlesize.
```

| @commercetools-frontend/assets/images/link-external\_\_working.svg | @commercetools-frontend/assets/images/camera-chain.svg        |
| @commercetools-frontend/assets/images/page-not-found.svg           | @commercetools-frontend/assets/images/desert-fox.svg          |
| @commercetools-frontend/assets/images/project-expired.svg          | @commercetools-frontend/assets/images/hourglass.svg           |
| @commercetools-frontend/assets/images/project-not-initialized.svg  | @commercetools-frontend/assets/images/hourglass.svg           |
| @commercetools-frontend/assets/images/project-suspended.svg        | @commercetools-frontend/assets/images/doors-closed.svg        |
| @commercetools-frontend/assets/images/unexpected-error.svg         | @commercetools-frontend/assets/images/icecream.svg            |

### Things removed

We removed some exports that are not necessary anymore:

- `application-components`:
  - `ServicePageResponseLayout` (was _deprecated_)
- `permissions`:
  - `hasPermission`
  - `permissions`
- `react-notifications`:
  - `GetCustomNotificationComponent`

### Dropped support for IE11

> We don't support IE 11 anymore, therefore we removed polyfills and things related to IE 11 support.

Reviewer: ```suggestion
Due to IE11's low usage and a high penality on mondern browsers due to transpilation and polyfills, we decided not to support IE 11 anymore. As a result polyfills and other things related such as transpilation were removed or adjusted.
```
